conda/plan.py

```diff
@@ -689,33 +691,55 @@ def get_actions_for_dists(dists_for_prefix, only_names, index, force, always_cop
 
 
 def augment_specs(prefix, specs, pinned=True):
-    # get conda-meta/pinned
+    """
+    Include additional specs for conda and (optionally) pinned packages.
+
+    Parameters
+    ----------
+    prefix : str
+        Environment prefix.
+    specs : list of MatchSpec
+        List of package specifications to augment.
+    pinned : bool, optional
+        Optionally include pinned specs for the current environment.
+
+    Returns
+    -------
+    augmented_specs : list of MatchSpec
+       List of augmented package specifications.
+    """
+    specs = list(specs)
+
+    # Get conda-meta/pinned
     if pinned:
         pinned_specs = get_pinned_specs(prefix)
         log.debug("Pinned specs=%s", pinned_specs)
-        specs += [MatchSpec(spec) for spec in pinned_specs]
+        specs += list(pinned_specs)
 
-    # support aggressive auto-update conda
+    # Support aggressive auto-update conda
     #   Only add a conda spec if conda and conda-env are not in the specs.
     #   Also skip this step if we're offline.
     root_only = ('conda', 'conda-env')
-    mss = [MatchSpec(s) for s in specs if s.name.startswith(root_only)]
+    mss = [spec for spec in specs if spec.name.startswith(root_only)]
     mss = [ms for ms in mss if ms.name in root_only]
+
     if is_root_prefix(prefix):
         if context.auto_update_conda and not context.offline and not mss:
             specs.append(MatchSpec('conda'))
             specs.append(MatchSpec('conda-env'))
     elif basename(prefix).startswith('_'):
-        # anything (including conda) can be installed into environments
+        # Anything (including conda) can be installed into environments
         # starting with '_', mainly to allow conda-build to build conda
         pass
     elif mss:
```

Comment:
Since we're cleaning this up...  I think this is the only line where `mss` is actually used, and the logic for creating it is really messed up.  I think lines 702/723 and 724 do exactly the same thing.  Plus, the `.startswith(root_only)` when the value of `root_only` is `('conda', 'conda-env')` 🤦‍♂️ .

So remove all this `mss` stuff, and just make line 712/734

    elif any(spec for spec in specs if spec.name.startswith('conda')):